### ISSUE TITLE
argparse allows nargs>1 for positional arguments but doesn't allow metavar to be a tuple

### ISSUE THREAD

**Issue by ccda776a-4736-4dbd-bf35-88e58f6952ba** [python/cpython#58282]
BPO | [14074](https://bugs.python.org/issue14074)
--- | :---
Nosy | @rhettinger, @cykerway, @shihai1991, @iritkatriel
PRs | <li>python/cpython#10847</li>
Files | <li>[issue14074.patch](https://bugs.python.org/file31042/issue14074.patch "Uploaded as text/plain at 2013-07-26.20:20:49 by paul.j3")</li><li>[issue14074_1.patch](https://bugs.python.org/file35049/issue14074_1.patch "Uploaded as text/plain at 2014-04-27.06:24:18 by paul.j3")</li><li>[14074.patch](https://bugs.python.org/file47965/14074.patch "Uploaded as text/plain at 2018-12-01.13:50:49 by @cykerway"): lightweight patch</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2012-02-21.17:38:38.183>
labels = ['type-bug', 'library', '3.9', '3.10', '3.11']
title = "argparse allows nargs>1 for positional arguments but doesn't allow metavar to be a tuple"
updated_at = <Date 2021-11-26.23:42:00.635>
user = 'https://bugs.python.org/tshepang'
```

bugs.python.org fields:
```python
activity = <Date 2021-11-26.23:42:00.635>
actor = 'iritkatriel'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2012-02-21.17:38:38.183>
creator = 'tshepang'
dependencies = []
files = ['31042', '35049', '47965']
hgrepos = []
issue_num = 14074
keywords = ['patch']
message_count = 13.0
messages = ['153884', '154236', '193737', '217184', '217240', '286227', '330854', '360841', '361154', '361184', '361215', '361336', '407109']
nosy_count = 8.0
nosy_names = ['rhettinger', 'bethard', 'tshepang', 'paul.j3', 'vvas', 'cykerway', 'shihai1991', 'iritkatriel']
pr_nums = ['10847']
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue14074'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

### Linked PRs
* gh-124782
* gh-124881
* gh-124882

**Comment by kkarbowiak** [comment]
Hi All.

I have stumbled upon this issue almost three years ago and [wrote a question](https://stackoverflow.com/questions/58541460/tuple-metavar-value-for-positional-argument-with-nargs-1) on StackOverflow.

My sample program was
```python
import argparse

parser = argparse.ArgumentParser()
parser.add_argument('size', type=int, nargs=2, help='size', metavar=('w', 'h'))

args = parser.parse_args()
print(args)
```
When the program is called as `prog.py --help` it produces an error that varies between Python3 versions (I tried 3.5, 3.6, and 3.8) which can be `ValueError: too many values to unpack (expected 1)` or `TypeError: sequence item 0: expected str instance, tuple found`.

For optional arguments
```python
import argparse

parser = argparse.ArgumentParser()
parser.add_argument('--size', type=int, nargs=2, help='size', metavar=('w', 'h'))

args = parser.parse_args()
print(args)
```
all is fine
```
usage: prog.py [-h] [--size w h]

optional arguments:
  -h, --help  show this help message and exit
  --size w h  size
```

The [official documentation](https://docs.python.org/3/library/argparse.html#metavar) mentions using tuples in metavar, but only provides an example for optional arguments.

I would argue that positional arguments should behave the same and would rather see this fixed than banned.
